lib/Common/Memory/Recycler.cpp

```diff
@@ -4358,7 +4390,11 @@ BOOL
 Recycler::RequestConcurrentWrapperCallback()
 {
 #if ENABLE_CONCURRENT_GC
+#if ENABLE_ALLOCATIONS_DURING_CONCURRENT_SWEEP
+    Assert(!IsConcurrentExecutingState() && !IsConcurrentSweepState());
```

Comment:
Why this change? It looks like it's sufficient to just check for IsConcurrentExecutingState since that implies we're not in concurrent sweep state, no?

Reply:
Actually, I guess its a little more subtle- the question is, can this function get called if we're midway during concurrent sweep? And it sounds like the answer is no? Although in theory, that should not be a problem since this function can only be called during Mark/Rescan, so I guess broadening the assert is probably fine- and there's no reason to ifdef it here, IsConcurrentSweepState should do the right thing in both cases

---
In reply to: [148170498](https://github.com/Microsoft/ChakraCore/pull/4104#discussion_r148170498) [](ancestors = 148170498)

Reply:
good point. fixed.

---
In reply to: [148171246](https://github.com/Microsoft/ChakraCore/pull/4104#discussion_r148171246) [](ancestors = 148171246,148170498)